### PR TITLE
MGMT-18219: Update go version to 1.22

### PR DESCRIPTION
**Pull request by eliorerz**
/cc @eranco74 @tsorya

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8/go-toolset:1.19 as builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17 AS builder
```
Comment (carbonin) [inline Dockerfile:1]: No, we need to use only public images for the konflux integration

**Review by carbonin — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (carbonin) [reply Dockerfile:1]: (I just changed this in https://github.com/openshift/image-based-install-operator/pull/70)

Comment (carbonin) [reply Dockerfile:1]: This one https://github.com/openshift/image-based-install-operator/pull/77/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R19

Comment (tsorya) [reply Dockerfile:1]: I don't see registry.access.redhat.com/ubi8/go-toolset:1.22

Comment (eliorerz) [reply Dockerfile:1]: The reason we need >= 1.22 we need `DataImage` from github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1 that requires go >= 1.22
`go: github.com/openshift/baremetal-operator/apis@v0.0.0-20240611081634-a92cc4b9ae8b requires go >= 1.22; switching to go1.22.6`

**Comment by carbonin** [comment]
Please also make sure you run `make bundle` and commit the changes if there's anything other than the timestamp.

**Comment by carbonin** [comment]
This was my concern :worried: 

```
Pulling image brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22 ...
Trying to pull brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22...
Warning: Pull failed, retrying in 5s ...
Trying to pull brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22...
Warning: Pull failed, retrying in 5s ...
Trying to pull brew.registry.redhat.io/rh-osbs/openshift-golang-builder:v1.22...
Warning: Pull failed, retrying in 5s ...
```

**Pull request merged**